src/pages/HomePage.jsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { setProducts } from '../redux/products/productSlice';
+import demoData from '../demo/demo.json'; 
+
+const HomePage = () => {
+  const dispatch = useDispatch();
+  const products = useSelector((state) => state.products.products);
+  // useEffect(() => {
+  //   const fetchData = async () => {
+  //     try {
+  //       const response = await fetch('demo/demo.json');
+  //       const data = await response.json();
+  //       dispatch(setProducts(data)); 
+  //     } catch (error) {
+  //       console.error('Error fetching data:', error);
+  //     }
+  //   };
+
+  //   fetchData();
+  // }, [dispatch]);
+
+  useEffect(() => {
+    dispatch(setProducts(demoData));
+  }, [dispatch]);
+
+  return (
+    <div className='pro-list'>
+      <h1>Product List</h1>
+      <ul className='products'>
+        {products.map((product) => (
+          <li key={product.id} className='product'>
+            <img src={product.picture} alt={product.name}/>
+            <h2>{product.name}</h2>
+            <button>View Details</button>
```

Comment:
I think this can be made into its own component for better readability. What do you think?

Reply:
homepage will empty